rules/engine.go

```diff
@@ -69,7 +70,7 @@ type V3Engine interface {
 	AddRule(rule DynamicRule,
 		lockPattern string,
 		callback V3RuleTaskCallback,
-		options ...RuleOption)
+		options ...RuleOption) error
 	AddPolling(namespacePattern string,
```

Comment:
I think we might also have to check the AddPolling namespacePattern, but have not looked into that one yet.

Reply:
AddPolling calls AddRule so I think it should be okay unless the check should be higher up in the chain https://github.com/IBM-Cloud/go-etcd-rules/blob/298d3d3d54676cbcef73c74e3eacaada124aadd4/rules/engine.go#L247